app/models/remote_theme.rb

```diff
@@ -417,12 +417,17 @@ def normalize_override(hex)
   end
 
   def update_theme_color_schemes(theme, schemes)
-    missing_scheme_names = Hash[*theme.color_schemes.pluck(:name, :id).flatten]
+    existing_schemes = ColorScheme.unscoped.where(theme_id: theme.id)
+
+    missing_scheme_names =
+      Hash[*existing_schemes.reject(&:remote_copy).map { |cs| [cs.name, cs] }.flatten]
```

Comment:
```suggestion
    missing_scheme_names =
      existing_schemes.reduce({}) do |hash, cs|
        hash[cs.name] = cs if !cs.remote_copy
        hash
      end
```

I believe we can do something simple like this which I personally find to be more readable. We also get to reduce the number of method calls and array allocations.